plugins/modules/mso_schema_site_bd_l3out.py

```diff
@@ -239,14 +237,14 @@ def main():
             ops.append(dict(op="replace", path="{0}/{1}".format(bd_path, bd), value=mso_objects.get("site_bd").details))
 
     elif state == "present":
-        if not payload:
+        if payload:
+            ops.append(dict(op="add", path="{0}/-".format(bd_path), value=payload))
+            mso.existing = mso.make_reference(l3out, "l3out", l3out_schema_id, l3out_template)
+        elif not mso.existing:
             l3outs.append(l3out.get("name"))
             l3out_refs.append(l3out_ref)
             ops.append(dict(op="replace", path="{0}/{1}".format(bd_path, bd), value=mso_objects.get("site_bd").details))
-            mso.existing = mso.make_reference(l3out, "l3out", mso_schema.id, template)
-        elif not mso.existing:
-            ops.append(dict(op="add", path="{0}/-".format(bd_path), value=payload))
-            mso.existing = mso.make_reference(l3out, "l3out", mso_schema.id, template)
+            mso.existing = mso.make_reference(l3out, "l3out", l3out_schema_id, l3out_template)
```

Comment:
Can we specify this-> mso.existing = mso.make_reference(l3out, "l3out", l3out_schema_id, l3out_template)
only once?

Reply:
agree have changed